packages/mui-lab/src/DesktopDatePicker/DesktopDatePicker.test.tsx

```diff
@@ -267,6 +267,27 @@ describe('<DesktopDatePicker />', () => {
 
     expect(screen.getByMuiTest('picker-toolbar')).toBeVisible();
   });
+  it('prop `clearable` - renders clear button in Desktop mode', () => {
+    const onChangeMock = spy();
+    render(
+      <UncontrolledOpenDesktopDatePicker
+        // @ts-expect-error internal prop
+        defaultOpen
+        showToolbar
+        clearable
+        TransitionComponent={FakeTransitionComponent}
+        value={adapterToUse.date('2018-01-01T00:00:00.000')}
+        renderInput={(params) => <TextField {...params} />}
+        onChange={onChangeMock}
+      />,
+    );
+
+    fireEvent.click(screen.getByText('Clear'));
+
+    expect(onChangeMock.callCount).to.equal(1);
+    expect(onChangeMock.args[0][0]).to.equal(null);
+    expect(screen.queryByRole('dialog')).to.equal(null);
```

Comment:
Shouldn't we also test that the value is actually cleared?

Reply:
Looks like this is the only comment left to be resolved

Reply:
I think it is not necessary to check the value, because in this case we would need to change the `texbox` value directly by something like: 
`fireEvent.change(screen.getByRole('textbox'), {
        target: { value: onChangeMock.args[0][0] },
      });`

and then we can check if the textbox value is cleared. I think it might not be valuable for this test as we have already checked the onChange arguments and callCount.

Reply:
The test should test as close as possible what the users are doing. If we want to test a clearable button, we should enter a date, fire click event on the clearable button and then expect that the value is cleared. We could add a default value in the date picker maybe at the start of the test?

Reply:
Yes sure. I'll add it

Reply:
I changed the test as you mentioned to replicate the user interaction more accurately. Now it'll check whether the dialog is closed and the textbox is cleared when the Clear button is triggered.